nevergrad/functions/control/core.py

```diff
@@ -39,19 +39,29 @@ class BaseFunction(ExperimentFunction):
         random state for reproducibility in Gym environment.
     """
 
-    def __init__(self, num_rollouts: int, random_state: tp.Optional[int] = None) -> None:
-        super().__init__(self._simulate, p.Array(shape=self.policy_dim))
+    def __init__(self, num_rollouts: int, activation: str = "tanh", random_state: tp.Optional[int] = None) -> None:
+        list_parametrizations = [p.Array(shape=(a, b)).set_name(r"layer_{a}_{b}") for a, b in
+                                 zip(self.policy_dim[:-1], self.policy_dim[1:])]
+        parametrization = p.Instrumentation(*list_parametrizations).set_name(self.env_name)
+        super().__init__(self._simulate, parametrization)
         self.num_rollouts = num_rollouts
         self.random_state = random_state
+        self.activation = activation
+        self.add_descriptors(num_rollouts=num_rollouts)
+        if not hasattr(self, "layer_rescaling_coef"): self.layer_rescaling_coef = np.ones(len(self.policy_dim) - 1)
         self._descriptors.pop("random_state", None)  # remove it from automatically added descriptors
 
-    def _simulate(self, x: np.ndarray) -> float:
+    def _simulate(self, x: tp.Tuple) -> float:
         env = GenericMujocoEnv(env_name=self.env_name,
-                               state_mean=self.state_mean,
-                               state_std=self.state_std,
+                               state_mean=self.state_mean if len(self.policy_dim) == 2 else None,
```

Comment:
This I don't understand... why should the presence of a neural net interfere with the use of this normalization ?

Reply:
You are right, it shouldn't